### PR TITLE
NOX-589 - Add new NoxType Extensions for Country and Money

### PR DESCRIPTION
**Pull request by dejobratic**
1. Add to Nox.Type.Country Extensions a new Extension that From a Nox.Reference.Country we can get the Nox.Type Country

2. Add to Nox.Type.Country Extensions a new Extension that From a Nox.Reference.Country.Id we can get the Nox.Type Country

3. Add to Nox.Type.Money Extensions a new Extension that From a Nox.Reference.Currency we can get the Nox.Types.CurrencyCode

### CODE REVIEW

File: samples/Cryptocash.Ui/Nox.Generated/Generator.g.cs
Comment (gjorgji-grgovski-authoritypartners) [inline samples/Cryptocash.Ui/Nox.Generated/Generator.g.cs:1]: Looks like ApplicationDto is marked as required and it is failing now the generator on UI, nothing to do with this PR, but @rochar  FYI.

File: tests/Nox.Yaml.Tests/TestDesigns/Nox/Schemas/domainCommand.json [hunk 1]
```diff
@@ -1903,20 +1903,7 @@
...
-                            "docx",
-                            "eml",
-                            "pdf",
-                            "ppt",
-                            "pptx",
-                            "txt",
-                            "xls",
-                            "xlsx"
-                          ]
+                          "type": "array"
```
Comment (gjorgji-grgovski-authoritypartners) [inline tests/Nox.Yaml.Tests/TestDesigns/Nox/Schemas/domainCommand.json:1906]: Why is this removed again, do you know @dejobratic ?

File: src/Nox.Types.Extensions/Types/CountryCode2/CountryExtensions.cs
Comment (rochar) [inline src/Nox.Types.Extensions/Types/CountryCode2/CountryExtensions.cs:1]: @dejobratic  to me we should only have one single CountryExtensions  static class, we are being to fine grain... what do y think?

this adds just complexity for a simple code

**Review by rochar — DISMISSED** [review]

File: src/Nox.Types.Extensions/Types/CountryCode2/CountryExtensions.cs
Comment (dejobratic) [reply src/Nox.Types.Extensions/Types/CountryCode2/CountryExtensions.cs:1]: Yes we can. Can we decide where it should be placed? I've split the class into multiple partials since we don't have a single Country class in Nox.Types, but 3 of them Iso2, Iso3 and Numeric codes.

File: tests/Nox.Yaml.Tests/TestDesigns/Nox/Schemas/domainCommand.json (hunk 1 above)
Comment (dejobratic) [reply tests/Nox.Yaml.Tests/TestDesigns/Nox/Schemas/domainCommand.json:1906]: I think it's related to this PR: https://github.com/NoxOrg/Nox/pull/1352.

**Review by gjorgji-grgovski-authoritypartners — DISMISSED** [review]

**Review by gjorgji-grgovski-authoritypartners — DISMISSED** [review]

File: src/Nox.Types.Extensions/Types/CountryCode2/CountryExtensions.cs
Comment (rochar) [reply src/Nox.Types.Extensions/Types/CountryCode2/CountryExtensions.cs:1]: @dejobratic  see my last commit on this PR, I would purpose to be like that, are y ok with this? Keep it simple, this for now should be very simple.

**Review by rochar — DISMISSED** [review]

File: src/Nox.Types.Extensions/Types/CountryCode2/CountryExtensions.cs
Comment (dejobratic) [reply src/Nox.Types.Extensions/Types/CountryCode2/CountryExtensions.cs:1]: Looks good to me. Thanks
